BaristaSwift/Common/Common/Base/Store.swift

```diff
@@ -33,20 +33,27 @@ import SmartSync
 import PromiseKit
 
 public class Store<objectType: StoreProtocol> {
+    
+    private var soupInitialized : Bool = false
+    private var syncsInitialized : Bool = false
 
     private final let pageSize: UInt = 100
 
     private final let syncDownName: String = "syncDown_\(objectType.objectName)";
     
     private final let syncUpName: String = "syncUp_\(objectType.objectName)";
     
+    private final let soupName: String = objectType.objectName;
+    
     public let sqlQueryString: String = SFRestAPI.soqlQuery(withFields: objectType.createFields, sObject: objectType.objectName, whereClause: nil, groupBy: nil, having: nil, orderBy: [objectType.orderPath], limit: 100)!
     
     public let queryString: String = "SELECT \(objectType.selectFieldsString()) FROM {\(objectType.objectName)} WHERE {\(objectType.objectName):\(Record.Field.locallyDeleted.rawValue)} != 1 ORDER BY {\(objectType.objectName):\(objectType.orderPath)} ASC"
     
     public init() {
-        let soupName = objectType.objectName
-        
+        NotificationCenter.default.addObserver(self, selector: #selector(handleUserWillLogout), name: NSNotification.Name(rawValue: "SFNotificationUserWillLogout") /* kSFNotificationUserWillLogout */, object: nil)
```

Comment:
@trooper2013 What would be a clean way to reference `kSFNotificationUserWillLogout`.
Right now I'm doing `NSNotification.Name(rawValue: "SFNotificationUserWillLogout")`

Reply:
@wmathurin  you should be able to do this.
``` 
NotificationCenter.default.addObserver(self, selector: #selector(handleUserWillLogout), name: NSNotification.Name(kSFNotificationUserWillLogout), object: nil)
```